### ISSUE TITLE
`fallbackCJS` is not supported

### ISSUE THREAD

**Issue by dctalbot** [thomasbrodusch/vitest-fail-on-console#59]
## Bug description

I think something is amiss with the packaging configuration. 

#### Description

- I have a typical vitest setup file with this contents:

```
import failOnConsole from "vitest-fail-on-console"
failOnConsole()
```

- I do not have `{type: "module"}` in my `package.json`
- Changing the setup file to be named `setup.mts` did not help

#### Expected outcome

- no error

#### Actual outcome

- vitest errors with `TypeError: default is not a function` when using this package.

**Comment by thomasbrodusch** [comment]
Hey @dctalbot thanks for raising a new issue!

As you are the first one raising that kind of issue can you provide a bit more of context:
- your `package.json`
- some screenshots, error stacktrace, ...

Thanks for using vitest-fail-on-console :)

**Comment by dctalbot** [comment]
`vitest.config.mts`:
```ts
import path from "path"
import { defineConfig } from "vitest/config"

export default () => {
  return defineConfig({
    test: {
      setupFiles: ["./test/setupVitest.ts"],
      server: {
        deps: {
          fallbackCJS: true,
        },
      },
    },
    resolve: {
      alias: {
        src: path.resolve(__dirname, "./src"),
        "@test": path.resolve(__dirname, "./test"),
      },
    },
  })
}

```

`test/setupVitest.ts`:
```ts
import failOnConsole from "vitest-fail-on-console"

failOnConsole()
```

`console output`:

```console
➜  my-project ✗ pnpm vitest run src/ui/Menu/Menu.test.tsx

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Suites 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/ui/Menu/Menu.test.tsx [ src/ui/Menu/Menu.test.tsx ]
TypeError: default is not a function
 ❯ test/setupVitest.ts:3:1
      1| import failOnConsole from "vitest-fail-on-console"
      2| 
      3| failOnConsole()
       | ^
      4| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed (1)
      Tests  no tests
   Start at  16:48:04
   Duration  259ms (transform 17ms, setup 0ms, collect 0ms, tests 0ms, environment 0ms, prepare 60ms)
```


**versions:**
vite 5.3.5
vitest 2.0.5
vitest-fail-on-console 0.7.0
Node 18.18.2

**Comment by dctalbot** [comment]
I think the issue may be with `fallbackCJS: true` which is documented as:
> Try to guess the CJS version of a package when it's invalid ESM

Maybe the dist is detcted as invalid? Unfortunately, I need this setting as `true` in order to support other packages.

**Comment by thomasbrodusch** [comment]
Hi @dctalbot new fixing release `v0.7.1` published & available on NPM 👌
It has been fixed in commit https://github.com/thomasbrodusch/vitest-fail-on-console/commit/f45eecdd5b661e93b663c1b851798ed27eaad03bf, have a try and tell me if everything run smoothly as expected :)

**Comment by dctalbot** [comment]
Hey, thank you for looking at this issue! Unfortunately, `v0.7.1` still exhibits the same behavior for me. I was able to unblock myself by reworking some other dependencies and setting `fallbackCJS` to `false`. Personally, this is no longer an issue, but other people may still run into this. I updated the issue title to reflect this.

**Comment by charliematters** [comment]
I'm one of the people mentioned above. I've had to turn on `fallbackCJS` for dependency reasons currently beyond my control, and the latest version of vitest-fail-on-console (0.10.1) was throwing the above errors.

I fixed it by adding this to my vitest.config.ts file:

```js
server: { deps: { fallbackCJS: true, inline: ['vitest-fail-on-console'] } },
```
